app/models/sponsorship.rb

```diff
@@ -47,4 +47,16 @@ def validate_level_uniqueness
 
     errors.add(:level, "You can have only one sponsorship of #{METAL_LEVELS.join(', ')}")
   end
+
+  def validate_sponsorable_type
+    return if sponsorable_type.blank?
+
+    begin
+      sponsorable_type.constantize
+      true
+    rescue NameError
+      errors.add(:sponsorable_type, "is not a valid class name")
+      false
+    end
+  end
```

Comment:
I don't think we should use an exception for this but use `safe_constantize` instead which returns `nil` instead of raising:

```ruby
"invalid".constantize
# NameError: wrong constant name invalid
"invalid".safe_constantize
#=> nil
```


```suggestion
  return if sponsorable_type.safe_constantize
   
   errors.add(:sponsorable_type, "is not a valid class name")
```

Reply:
thanks for the hint - I'm moving this in another direction (allowing only Tag and its parent until someone finds a case to sponsor something else).

The initial PR to create the sponsorships indicated the polymorphic association was a future proofing, but it looks like we don't have a second type of thing to sponsor yet, and can add that in the code (or via some yet to be imagined admin interface) for the time being.

Reply:
as a follow up - the "sponsorable" things are model classes that show this (only Tag here)

```ruby
has_one :sponsorship, inverse_of: sponsorable
```